The Guardian.js

```diff
@@ -82,39 +92,67 @@ function scrape(doc, url) {
 	var translator = Zotero.loadTranslator('web');
 	// Embedded Metadata
 	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
-	//translator.setDocument(doc);
-	
+	translator.setDocument(doc);
+
 	translator.setHandler('itemDone', function (obj, item) {
-		//The authors in the metadata are incomplete and are not cleaned,
-		//but contain unneccessary data about location etc.
-		//Thus we try to take them directly from the byline.
-		var byline = ZU.xpathText(doc, '//p[contains(@class, "byline")]');
+		let authors = new Set();
+
+		for (let author of doc.querySelectorAll('a[rel="author"]')) {
+			authors.add(author.textContent);
+		}
+
+		// The authors in the metadata are incomplete and are not cleaned,
+		// but contain unneccessary data about location etc.
+		// Thus we try to take them directly from the byline.
+		let byline = text(doc, 'address[data-link-name="byline"] *');
 		if (byline) {
-			item.creators = [];
-			var authors = byline.replace(/\s(in|for)\s.+/, '').split(/\s+and\s+|\s*,\s*/);
-			for (var i=0; i<authors.length; i++) {
-				item.creators.push(ZU.cleanAuthor(authors[i], "author"));
+			// parse out linked publications
+			byline = byline.split(/ for /)[0];
+			// parse out location
+			byline = byline.split(/\s*\b(?:in|at)\b\s*/)[0];
+
+			// parse authors from rest of byline
+			for (let author of byline.split(/\s*(?:\band\b|,)\s*/)) {
+				authors.add(author);
 			}
 		}
-		item.language = "en-GB";
-		// og:url does not preserve https prefixes, so use canonical link until fixed
-		var canonical = doc.querySelector('link[rel="canonical"]');
-		if (canonical) {
-			item.url = canonical.href;
+
+		let linkedData = JSON.parse(text(doc, 'script[type="application/ld+json"]'));
+		if (linkedData instanceof Array) linkedData = linkedData.find(x => x["@type"] == "NewsArticle");
```

Comment:
`instanceof Array` [has some issues](https://web.mit.edu/jwalden/www/isArray.html), so it would be best to use `Array.isArray` here.